### PR TITLE
add an error for stack allocations in naked functions

### LINKED ISSUE #72: add error for stack allocation in naked functions
``` zig
#attribute("naked")
fn f() => {
    var a: i32; // error, no stack frame in naked functions
}
```

### PR DESCRIPTION
**Pull request by Rexicon226**
closes #72

### CODE REVIEW

File: test/cases/compile_errors/stack_usage_in_naked_function.zig [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    var x: Foo = .{ .y = 10 };
+    _ = &x;
+}
+
+// error
+// backend=stage2
+//
+// :2:12: error: no stack frame in naked functions
+// :10:9: error: no stack frame in naked functions
+// :23:12: error: no stack frame in naked functions
```
Comment (andrewrk) [inline test/cases/compile_errors/stack_usage_in_naked_function.zig:32]:
```suggestion
// :2:5: error: local variable in naked function
// :10:5: error: local variable in naked function
// :23:5: error: local variable in naked function
```

**Review by andrewrk — CHANGES_REQUESTED** [review]
The error message shouldn't mention the stack, only the language rule. Pointing at the type is a bit weird. Pointing at the keyword or name is OK.

**Comment by zeroZshadow** [comment]
Will this error also trigger for usage of `var` in an `inline` function that is called inside a `.Naked` function?

**Comment by andrewrk** [comment]
A good question that should be answered with a few more lines in the test case.

**Comment by andrewrk** [comment]
Thanks!

**Pull request merged**